### ISSUE TITLE
Something wrong with pytorch implement

### ISSUE THREAD

**Issue by eclickECNU** [harvardnlp/im2markup#19]
Hi,

I tried to implement this model with pytorch, but I encountered some problems. As mentioned in your paper(http://arxiv.org/pdf/1609.04938v1.pdf),experiments are run on a 12GB Nvidia Titan X GPU, you train the model for 12 epochs and use the validation perplexity to choose the best model.But in my experiment, the accuracy of 20 epochs was still 0. After that, I try to use a small number of training samples(10 samples) to train to check the correctness of the code. I found that the loss would converge untill 500 epochs. My GPU is GTX 1080Ti * 2, in the case of batch_size = 16, an epoch takes about 30 minutes(batch_size=20 will OOM). But even if 500 epoch can complete the training, this time is too long to bear. I checked my code repeatedly but didn't find other erros. I am particularly curious as to what the problem is and I don't understand why there is such a big difference. Can you provide the code for the pytorch version to learn? Or what mistakes do you think might cause this problem?

Looking forward to your replay

**Comment by Miffyli** [comment]
Those experiments were ran using [Torch library for Lua](http://torch.ch/), not with PyTorch.

**Comment by da03** [comment]
@eclickECNU have you managed to reproduce our results in pytorch yet? We have an implementation in PyTorch as part of OpenNMT-py: https://github.com/OpenNMT/OpenNMT-py/blob/master/docs/source/im2text.md

**Comment by eclickECNU** [comment]
> @eclickECNU have you managed to reproduce our results in pytorch yet? We have an implementation in PyTorch as part of OpenNMT-py: https://github.com/OpenNMT/OpenNMT-py/blob/master/docs/source/im2text.md

Thank you very much!

**Comment by msrend** [comment]
Is it possible for you to get your pytorch version of image to latex model @eclickECNU ?
I am not familiar with Lua and I really want to know how it works in pytorch. 
I implemented my pytorch model but it didnt seem to work well.
